libs/shacl/core/services/shacl-helper.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { DataModelDefinitionHelper, PrefixCcService } from '@cognizone/json-model';
+import { Many, manyToArray, manyToOne, notNil, TypedResourceContext } from '@cognizone/model-utils';
+
+import { ShaczShapesGraph, ShNodeShape, ShPropertyShape } from '../models';
+
+@Injectable({ providedIn: 'root' })
+export class ShaclHelper implements DataModelDefinitionHelper<ShaclHelperDefinition> {
+  constructor(private prefixCcService: PrefixCcService) {}
+
+  hasProperty(definition: ShaclHelperDefinition, type: Many<string>, propertyKey: string): boolean {
+    return !!this.getPropertyShape(definition, type, propertyKey);
+  }
+
+  isAttribute(definition: ShaclHelperDefinition, type: Many<string>, propertyKey: string): boolean {
+    const property = this.getPropertyShape(definition, type, propertyKey);
+    return !!property?.['sh:datatype'];
+  }
+
+  isReference(definition: ShaclHelperDefinition, type: Many<string>, propertyKey: string): boolean {
+    const property = this.getPropertyShape(definition, type, propertyKey);
+    return !!property?.['sh:class'];
+  }
+
+  isSingle(definition: ShaclHelperDefinition, type: Many<string>, propertyKey: string): boolean {
+    const property = this.getPropertyShape(definition, type, propertyKey);
+    return property?.['sh:maxCount'] === 1;
+  }
+
+  isRequired(definition: ShaclHelperDefinition, type: Many<string>, propertyKey: string): boolean {
+    const property = this.getPropertyShape(definition, type, propertyKey);
+    return property?.['sh:minCount'] === 1 && property?.['sh:maxCount'] === 1;
```

Comment:
I think `sh:minCount === 1` is enough here to fullfill the `isRequired` condition, no need to check maxCount